feat/tests/test_fast_detector.py

```diff
@@ -35,3 +44,225 @@ def test_detect_image(self, single_face_img):
+        out = self.detector.detect(
+            [single_face_img, no_face_img, multi_face_img] * 2,
+            batch_size=4,
+            output_size=300,
+        )
+        assert out.shape == (14, EXPECTED_FEX_WIDTH)
+       
+    # Single images    
+    def test_fast_detect_single_img_single_face(self, single_face_img):
+        """Test detection of single face from single image. Default detector returns EXPECTED_FEX_WIDTH attributes"""
+        out = self.detector.detect(single_face_img)
+        assert type(out) == Fex
+        assert out.shape == (1, EXPECTED_FEX_WIDTH)
+        assert out.happiness.values[0] > 0
+    
+    def test_fast_detect_single_img_multi_face(self, multi_face_img):
+        """Test detection of multiple faces from single image"""
+        out = self.detector.detect(multi_face_img)
+        assert type(out) == Fex
+        assert out.shape == (5, EXPECTED_FEX_WIDTH)
+        
+    def test_fast_detect_with_alpha(self):
+        image = os.path.join(get_test_data_path(), "Image_with_alpha.png")
+        out = self.detector.detect(image)
+        
+    # Multiple images
+    def test_fast_detect_multi_img_single_face(self, single_face_img):
+        """Test detection of single face from multiple images"""
+        out = self.detector.detect([single_face_img, single_face_img])
+        assert out.shape == (2, EXPECTED_FEX_WIDTH)
+
+
+    def test_fast_detect_multi_img_multi_face(self, multi_face_img):
+        """Test detection of multiple faces from multiple images"""
+        out = self.detector.detect([multi_face_img, multi_face_img])
+        assert out.shape == (10, EXPECTED_FEX_WIDTH)
+
+
+    def test_fast_detect_images_with_batching(self, single_face_img):
+        """Test if batching works by passing in more images than the default batch size"""
+
+        out = self.detector.detect([single_face_img] * 6, batch_size=5)
+        assert out.shape == (6, EXPECTED_FEX_WIDTH)
+
+
+    def test_fast_detect_mismatch_image_sizes(self, single_face_img, multi_face_img):
+        """Test detection on multiple images of different sizes with and without batching"""
+
+        out = self.detector.detect([multi_face_img, single_face_img])
+        assert out.shape == (6, EXPECTED_FEX_WIDTH)
+
+        out = self.detector.detect(
+            [multi_face_img, single_face_img] * 5, batch_size=5, output_size=512
+        )
+        assert out.shape == (30, EXPECTED_FEX_WIDTH)
+
+
+    def test_fast_detect_video(
+        self, single_face_mov, no_face_mov, face_noface_mov, noface_face_mov
+    ):
+        """Test detection on video file"""
+        out = self.detector.detect(single_face_mov, skip_frames=24, data_type="video")
+        assert len(out) == 3
+        assert out.happiness.values.max() > 0
+
+        # Test no face movie
+        out = self.detector.detect(no_face_mov, skip_frames=24, data_type="video")
```

Comment:
The same issue of us not handling missing faces it causing this test to fail as well. But the proposed fix (assuming no other issues) should solve this as well.